### PR TITLE
[FIX] Inconsistency In CI for Release Webhook

### LINKED ISSUE #295: [BUG] Sample platform can't handle release deletion
Please prefix your issue with one of the following: [BUG], [PROPOSAL], [QUESTION].

Sample platform commit (found at the bottom of each page) : Any

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [x] I have checked that the issue I'm posting isn't already reported.
- [x] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/sample-platform/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/sample-platform/issues)
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never visited/used the platform.
- [ ] I have used the platform just a couple of times.
- [ ] I have used the platform extensively, but have not contributed previously.
- [ ] I am an active contributor to the platform.
- [x] I am the BDFL of this project
---

Deleting a release causes a SQLAlchemy duplicate entry; when deleting/modifying a release there shouldn't be a new entry added.

### PR DESCRIPTION
**Pull request by thealphadollar**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

The PR allows for smoother operation in response to the trigger of `release` webhook.

- Delete CCEx version when a release is deleted or unpublished.
- Update CCEx version when a release is edited or published.
- We do not alter the existing tests since they are based on the latest commit and not release.

fixes #295

### CODE REVIEW

**Comment by canihavesomecoffee** [comment]
Please add additional unittests to cover the new code.

**Comment by thealphadollar** [comment]
@canihavesomecoffee Done.

File: tests/test_ci/TestControllers.py [hunk 1]
```diff
@@ -459,3 +504,27 @@ def generate_git_api_header(event, sig):
             ('Content-Type', 'application/json'),
             ('Content-Length', 6615)
         ])
+
+    def add_CCExversion(self, c, version='v2.1', commit='abcdefgh', released='2018-05-30T20:18:44Z'):
```
Comment (canihavesomecoffee) [inline tests/test_ci/TestControllers.py]: This seems contradictory... You say "add a verson to the DB", but you emulate a full POST request to the webhook... Why not just directly push something into the DB?

Comment (thealphadollar) [reply tests/test_ci/TestControllers.py]: My mistake; it just fell out of my mind. I've reverted to using direct insertion.

**Pull request merged**